src/openloco/window.cpp

```diff
@@ -18,6 +21,236 @@ namespace openloco::ui
+
+                    sub_45FD41(x, y, 128, gCurrentRotation, &outX, &outY, &outZ);
+                    viewport_set_underground_flag(this, viewport, false);
+
+                    bool atMapEdge = false;
+                    if (outX < -256)
+                    {
+                        outX = -256;
+                        atMapEdge = true;
+                    }
+                    if (outY < -256)
+                    {
+                        outY = -256;
+                        atMapEdge = true;
+                    }
+                    if (outX > 0x30FE)
+                    {
+                        outX = 0x30FE;
+                        atMapEdge = true;
+                    }
+                    if (outY > 0x30FE)
+                    {
+                        outY = 0x30FE;
+                        atMapEdge = true;
+                    }
+
+                    if (atMapEdge)
+                    {
+                        auto coord_2d = coordinate_3d_to_2d(ax, bx, 128, gCurrentRotation);
+
+                        this->saved_view_x = coord_2d.x - viewport->view_width / 2;
+                        this->saved_view_y = coord_2d.y - viewport->view_height / 2;
+                    }
+
+                    x2 = this->saved_view_x;
+                    y2 = this->saved_view_y;
+
+                    if (this->flags & window_flags::scrolling_to_location)
+                    {
+                        bool flippedX = false;
+                        x2 -= viewport->view_x;
+                        if (x2 < 0)
+                        {
+                            x2 = -x2;
+                            flippedX = true;
+                        }
+
+                        bool flippedY = false;
+                        y2 -= viewport->view_y;
+                        if (y2 < 0)
+                        {
+                            y2 = -y2;
+                            flippedY = true;
+                        }
+
+                        x2 = ceil(x2 / 8.0);
+                        y2 = ceil(y2 / 8.0);
+
+                        if (x2 == 0 && y2 == 0)
+                        {
+                            this->flags &= ~window_flags::scrolling_to_location;
+                        }
+
+                        if (flippedX)
+                        {
+                            x2 = -x2;
+                        }
+
+                        if (flippedY)
+                        {
+                            y2 = -y2;
+                        }
+
+                        x2 += viewport->view_x;
+                        y2 += viewport->view_y;
+                    }
+                }
+                viewport_move(x2, y2, this, viewport);
+            }
+        }
+
+        if (this->viewports[1] != nullptr)
+        {
+            this->call_on_resize();
+
+            viewport* viewport = this->viewports[0];
+            if (viewport != nullptr)
+            {
```

Comment:
Should there be code in this block?

Reply:
It's what the assembly does; first a null check, then call resize and then another null check. Though I guess the first one should be pretty safe to remove